tests/wpunit/NodeByUriTest.php

```diff
@@ -1948,4 +1948,45 @@ public function testQueryPostsPageByUriReturnsExpectedUriOnNode() {
 
 	}
 
+	public function testQueryPostBySlugWhenPermalinksAreSetCustom() {
+
+		$this->set_permalink_structure( '/articles/%postname%/' );
+
+		$slug = 'test-slug';
+
+		$postId = $this->factory()->post->create([
+			'post_type'   => 'post',
+			'post_status' => 'publish',
+			'post_name' => $slug,
+			'post_title' => 'post-title'
+ 		]);
```

Comment:
Do we want to create a second post here, just to confirm it's actually matching and not returning the most recent?

Reply:
@justlevine good call. updated the test.